app/lib/foreman_deployments/tasks/base_definition.rb

```diff
@@ -6,7 +6,7 @@ class BaseDefinition
 
       def initialize(params = {})
         @task_id = task_id
-        @parameters = params
+        @parameters = HashWithIndifferentAccess[params]
```

Comment:
I'm getting `ArgumentError: odd number of arguments for Hash` from the unit test on this line. The problem is yaml passes `nil` params for tagged nodes without any inner data. I think we should change the constructor to:
```ruby
def initialize(params = nil)
  params ||= {}
  # ...
```
...or similar functionality.